File: MatMan13_16x64/MatMan_VB/AppCode/ApplicationServer/ThreadCultureSwitch.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading
Imports System.Threading.Tasks

Public Class ThreadCultureSwitch

    Public Property ThisThread() As Thread
	Get
		Return m_ThisThread
	End Get
	Set
		m_ThisThread = Value
	End Set
    End Property
    Private m_ThisThread As Thread

    Public Property OriginalCulture() As System.Globalization.CultureInfo
	    Get
		    Return m_OriginalCulture
	    End Get
	    Set
		    m_OriginalCulture = Value
	    End Set
    End Property
    Private m_OriginalCulture As System.Globalization.CultureInfo

    Public Sub New()
	    ThisThread = Thread.CurrentThread
	    OriginalCulture = ThisThread.CurrentCulture
    End Sub

    Public Sub SetUSCulture()
	    ThisThread.CurrentCulture = New System.Globalization.CultureInfo("en-US")
    End Sub

    Public Sub SetOriginalCulture()
	    ThisThread.CurrentCulture = Me.OriginalCulture
    End Sub

End Class

File: MatMan13_16x64/MatMan_Lib/AppCode/Application Server/ThreadCultureSwitch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace iiiwave.MatManLib
{
    public class ThreadCultureSwitch
    {
	    public Thread ThisThread                                { get; set; }
	    public System.Globalization.CultureInfo OriginalCulture { get; set; }

	    public ThreadCultureSwitch()
	    {
		    ThisThread      = Thread.CurrentThread;
		    OriginalCulture = ThisThread.CurrentCulture;
	    }

	    public void SetUSCulture()
	    {
		    ThisThread.CurrentCulture = new System.Globalization.CultureInfo("en-US");
	    }

	    public void SetOriginalCulture()
	    {
		    ThisThread.CurrentCulture = this.OriginalCulture;
	    }

    }
}
